tests/densify/test_interface_patchmatchnet.py

```diff
@@ -0,0 +1,108 @@
+"""Unit tests for the interface for PatchmatchNet
+
+Authors: Ren Liu
+"""
+import unittest
+from pathlib import Path
+
+import numpy as np
+from gtsam import PinholeCameraCal3Bundler
+
+from gtsfm.common.gtsfm_data import GtsfmData, SfmTrack
+from gtsfm.loader.olsson_loader import OlssonLoader
+from gtsfm.densify.interface_patchmatchnet import PatchmatchNetData
+
+DATA_ROOT_PATH = Path(__file__).resolve().parent.parent / "data"
+
+DEFAULT_FOLDER = DATA_ROOT_PATH / "set1_lund_door"
+
+NUM_VIEWS = 5
+
+EXAMPLE_CAMERA_ID = 1
+
+
+class TestInterfacePatchmatchNet(unittest.TestCase):
+    """Unit tests for the interface for PatchmatchNet."""
+
+    def setUp(self) -> None:
+        """Set up the image dictionary and gtsfm result for the test."""
+        super().setUp()
+
+        # initialize Olsson Loader
+        loader = OlssonLoader(str(DEFAULT_FOLDER), image_extension="JPG")
+        self._num_images = len(loader)
+
+        # set up image dictionary
+        self._img_dict = {i: loader.get_image(i) for i in range(self._num_images)}
+
+        # initialize an empty sfm result in GtsfmData class
+        self._sfm_result = GtsfmData(self._num_images)
+
+        # random valid cameras, the below number and indices of valid cameras are dummy values
+        self._num_valid_cameras = self._num_images // 2
+        valid_cameras = np.arange(self._num_valid_cameras)
+
+        # add dummy valid cameras to sfm result
+        for i in valid_cameras:
+            self._sfm_result.add_camera(
+                i, PinholeCameraCal3Bundler(loader.get_camera_pose(i), loader.get_camera_intrinsics(i))
+            )
```

Comment:
do we want to only add some cameras? If yes, the comment above this line can be made better.